asterisk/agi/library/Agi/Action/FaxCallAction.php

```diff
@@ -188,30 +188,67 @@ public function processFaxInStatus()
         // TODO Check return value for errors (use exec instead of shell_exec?)
         // Remove received tif file after conversion
         unlink($faxTIF);
+
+        // Store fax pages
+        $faxIn->setPages($this->agi->getVariable("FAXOPT(pages)"));
+
         // Check if this fax is associated with an email address
         if ($fax->getSendByEmail()) {
             $this->agi->verbose("Sending Fax [faxInOut%d] to %s", $faxIn->getId(), $fax->getEmail());
 
-            // Get faxdata values for mail message body and subject
-            $faxSrc = $faxIn->getSrc();
-            $faxDst = $faxIn->getDst();
+            // Get Fax Company
+            $company = $fax->getCompany();
+
+            // Get defaults mail settings
+            $config = \Zend_Controller_Front::getInstance()->getParam('bootstrap');
+            $mail = $config->getOption('mail');
+            $default_fromname = $mail['fromname'];
+            $default_fromuser = $mail['fromuser'];
+            $voicemail = $config->getOption('faxmail');
+            $template = $voicemail['template'];
 
             // Create attachment for PDF file
+            $pdfname = sprintf("fax-%s-%s.pdf", $faxIn->getSrc(), $faxIn->getDst());
             $attach = new \Zend_Mime_Part(file_get_contents($faxPDF));
             $attach->type = "application/pdf";
             $attach->disposition = \Zend_Mime::DISPOSITION_ATTACHMENT;
             $attach->encoding = \Zend_Mime::ENCODING_BASE64;
-            $attach->filename = "fax-$faxSrc-$faxDst.pdf";
+            $attach->filename = $pdfname;
+
+            $fromName = $company->getBrand()->getFromName();
+            if (empty($fromName)) $fromName = $default_fromname;
```

Comment:
Sure! This was copy-_pasta_ from voicemail notifications where it's also inline hehe